packages/mock-block-dock/src/hook-portals/text.tsx

```diff
@@ -103,10 +116,16 @@ export const TextHookView = ({
     [text],
   );
 
-  const textHasChanged = useDidTextChange(text);
-  if (textHasChanged) {
-    resetNodes(editor, nodesFromProps);
-  }
+  useEffect(() => {
+    if (
+      generateComparableString(text) !==
+      generateComparableString(previousTextRef.current)
+    ) {
+      resetNodes(editor, nodesFromProps);
+
+      previousTextRef.current = generateComparableString(text);
+    }
```

Comment:
You're calling `generateComparableString` on `previousTextRef.current` even though it already stores a comparable string, and you're doing the work twice. 

```suggestion
   const comparableString = generateComparableString(text);
    if (comparableString !== previousTextRef.current) {
      previousTextRef.current = comparableString;
      resetNodes(editor, nodesFromProps);
    }
```